packages/headless/src/features/folding/folding-slice.ts

```diff
@@ -141,4 +171,53 @@ export const foldingReducer = createReducer(
                 payload.numberOfFoldedResults ?? state.numberOfFoldedResults,
             }
       )
+      .addCase(loadAll.pending, (state, {meta}) => {
+        const collectionId = meta.arg;
+
+        return {
+          ...state,
+          collections: state.collections.map((collection) =>
```

Comment:
Should collections be stored as an array or as a dictionary where the key is the collection id?
I don't think collections are ordered so an array does not feel like the right data structure?

Reply:
Collections are ordered since they are search results, but from a store point of view I think it makes sense to use a dictionary as you say since the results list already contains the order.